components/fxa-client/src/device.rs

```diff
@@ -250,3 +203,13 @@ pub struct AttachedClient {
     pub last_access_time: Option<i64>,
     pub scope: Option<Vec<String>>,
 }
+
+/// Device record information for the client.  This is passed in at startup.
```

Comment:
Unless I'm missing something, isn't the dilemma here going to be that the mobile apps fetching the device record from local storage might be non-trivial - you might have managed to avoid IO in the FxA constructor, but that's not much help if it's mandatory that consumers of FxA must do IO before using it.

Reply:
I thought that they could just construct a local device record without IO, but I could definitely be wrong about that.   If I am, then maybe this should be part of the `SavedState` data that the `FxaStorage` callback interface loads.

Reply:
Yeah, most of the info comes from the server so IIUC is persisted locally. I agree it's probably best as part of SavedState.